t3/simulate/rms_simulator.py

```diff
@@ -0,0 +1,232 @@
+        if self.rms_reactor_type == 'ConstantVDomain':
+            self.domain, self.y0, self.p = rms.ConstantVDomain(phase=self.phase, initialconds=self.initialconds)
+        elif self.rms_reactor_type == 'ConstantPDomain':
+            self.domain, self.y0, self.p = rms.ConstantPDomain(phase=self.phase, initialconds=self.initialconds)
+        elif self.rms_reactor_type == 'ConstantTPDomain':
+            self.domain, self.y0, self.p = rms.ConstantTPDomain(phase=self.phase, initialconds=self.initialconds)
+
+
+    def simulate(self):
+        """
+        Simulate the mechanism with RMS
+        """
+        self.logger.info('Simulating a job using RMS...')
+
+        solver = de.CVODE_BDF()
+        t_final = convert_termination_time_to_seconds(self.rmg['reactors'][0]['termination_time'])
+
+        if len(self.observable_list):
+            react = rms.Reactor(self.domain, self.y0, (0.0, t_final), forwardsensitivities=True, p=self.p)
+            atol = [self.atol] * self.domain.indexes[-1]  # set atol for the variables being simulated
+            total_variables = self.domain.indexes[-1] + len(self.p) * self.domain.indexes[-1]
+            atol.extend([self.sa_atol] * (total_variables - self.domain.indexes[-1]))  # set atol for SA
+            self.sol = de.solve(react.ode, solver, abstol=atol, reltol=self.rtol)
+        else:
+            react = rms.Reactor(self.domain, self.y0, (0.0, t_final), forwardsensitivities=False, p=self.p)
+            self.sol = de.solve(react.ode, solver, abstol=self.atol, reltol=self.rtol)
+        self.bsol = rms.Simulation(self.sol, self.domain)
+
+
+    def get_sa_coefficients(self):
+        """
+        Obtain the SA coefficients.
+
+        Returns:
+             sa_dict: a SA dictionary, whose structure is given in the docstring for main.py
+        """
+        sa_dict = {'thermo': dict(), 'kinetics': dict(), 'time': list(self.sol.t)}
+        species_labels = [spc.name for spc in self.spcs]
+        num_rxns = len(self.rxns)
+        for observable in self.observable_list:
+            # get thermo SA coefficients
+            sa_dict['thermo'][observable] = dict()
+            for label in species_labels:
+                sa_dict['thermo'][observable][label] = list()
+                for t in sa_dict['time']:
+                    sa_dict['thermo'][observable][label].append(rms.getconcentrationsensitivity(
+                        self.bsol, observable, label, t))
+
+            # get kinetics SA coefficients
+            sa_dict['kinetics'][observable] = dict()
+            for rxn_number in range(num_rxns):
+                sa_dict['kinetics'][observable][rxn_number] = list()
+                for t in sa_dict['time']:
+                    sa_dict['kinetics'][observable][rxn_number].append(rms.getconcentrationsensitivity(
+                        self.bsol, observable, rxn_number, t))
+        return sa_dict
+
+
+    def get_idt(self):
```

Comment:
I'll update the function name. I had added these just because it was easy to do with Cantera. T3 doesn't use these methods, so I hadn't added them to the factory.